gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/util/JobStateUtils.java

```diff
@@ -71,23 +104,69 @@ public static StateStore<TaskState> openTaskStateStoreUncached(JobState jobState
     return new FsStateStore<>(fs, taskStateStorePath.toUri().getPath(), TaskState.class);
   }
 
+  /**
+   * ATTENTION: derives path according to {@link org.apache.gobblin.runtime.mapreduce.MRJobLauncher} conventions, using same
+   * {@link ConfigurationKeys#MR_JOB_ROOT_DIR_KEY}
+   * @return "base" dir root path for work dir (parent of inputs, output task states, etc.)
+   */
+  public static Path getWorkDirRoot(JobState jobState) {
+    return new Path(
+        new Path(jobState.getProp(ConfigurationKeys.MR_JOB_ROOT_DIR_KEY), jobState.getJobName()),
+        jobState.getJobId());
+  }
+
```

Comment:
Might make sense in the future for either the MR job launcher or a MR Job Launcher util to have this class since it's deeply tied to the MR implementation

Reply:
agreed we could have it live somewhere else.  although it originated w/ MR, the conventions are reasonable and well-organized enough that it can really be used far more widely than MR.  e.g. it seems perfectly reasonable for us to adopt as well for temporal.  if you agree, we might even name that future utils class w/o mentioning MR in the name